src/pages/PlanPage/subRoutes/UpgradePlanPage/UpgradeForm/Controllers/TeamPlanController/TeamPlanController.tsx

```diff
@@ -1,30 +1,100 @@
-import { FieldValues, UseFormRegister } from 'react-hook-form'
+import { UseFormRegister, UseFormSetValue } from 'react-hook-form'
+import { useParams } from 'react-router-dom'
 
-import { MIN_NB_SEATS_PRO } from 'shared/utils/upgradeForm'
+import {
+  IndividualPlan,
+  useAccountDetails,
+  useAvailablePlans,
+} from 'services/account'
+import {
+  canApplySentryUpgrade,
+  findProPlans,
+  findSentryPlans,
+} from 'shared/utils/billing'
+import {
+  MIN_NB_SEATS_PRO,
+  TEAM_PLAN_MAX_ACTIVE_USERS,
+  UPGRADE_FORM_TOO_MANY_SEATS_MESSAGE,
+} from 'shared/utils/upgradeForm'
 import TextInput from 'ui/TextInput'
 
 import BillingOptions from './BillingOptions'
 import PriceCallout from './PriceCallout'
 import UserCount from './UserCount'
 
 import { NewPlanType } from '../../constants'
+import { UpgradeFormFields } from '../../UpgradeForm'
+
+interface Errors {
+  seats?: {
+    message?: string
+  }
+}
+
+interface ErrorBannerProps {
+  errors?: Errors
+  setFormValue: UseFormSetValue<UpgradeFormFields>
+  setSelectedPlan: (plan: IndividualPlan) => void
+}
+
+function ErrorBanner({
+  errors,
+  setFormValue,
+  setSelectedPlan,
+}: ErrorBannerProps) {
+  const { provider, owner } = useParams<{ provider: string; owner: string }>()
+  const { data: plans } = useAvailablePlans({ provider, owner })
+  const { data: accountDetails } = useAccountDetails({ provider, owner })
+  const { proPlanYear } = findProPlans({ plans })
+  const { sentryPlanYear } = findSentryPlans({ plans })
+  const plan = accountDetails?.rootOrganization?.plan ?? accountDetails?.plan
+  const isSentryUpgrade = canApplySentryUpgrade({ plan, plans })
+  const yearlyProPlan = isSentryUpgrade ? sentryPlanYear : proPlanYear
+
+  if (!errors?.seats?.message) {
```

Comment:
Maybe this should be pulled out to where we're instatiating the errorBoundary on 127, just so it's a bit clearer to the next reader that the ErrorBanner isn't always present.